courses/views.py

```diff
@@ -38,6 +41,46 @@ class ProgramViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = ProgramSerializer
 
 
+class ProgramLearnersView(APIView):
+    """API for Learners enrolled in the Program"""
+
+    authentication_classes = (
+        SessionAuthentication,
+        TokenAuthentication,
+    )
+    permission_classes = (
+        IsAuthenticated,
+    )
+    serializer_class = ProfileImageSerializer
+
+    def get(self, request, *args, **kargs):
+        """
+        Get eight random learners with images and
+        the total count of visible learners in the program
+        """
+        program_id = self.kwargs["program_id"]
+        users = ProgramEnrollment.objects.filter(
+            program_id=program_id
+        ).values_list('user', flat=True)
+
+        queryset = Profile.objects.exclude(
+            image_small__exact=''
+        ).filter(user__in=users).exclude(
+            account_privacy='private'
+        ).exclude(
+            user=request.user
+        ).order_by('?')
```

Comment:
I don't super understand how querysets work, but will this run two separate queries, one to get the user list (`users` on line 62 above) and then another here to get the profiles? is there any way to do it in one?